common/src/main/java/com/wynntils/features/chat/GuildRankReplacementFeature.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.chat;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.text.PartStyle;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.core.text.StyledTextPart;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.models.players.GuildModel;
+import com.wynntils.utils.type.IterationDecision;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.Style;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@ConfigCategory(Category.CHAT)
+public class GuildRankReplacementFeature extends Feature {
+    @RegisterConfig
+    public final Config<RankType> rankType = new Config<>(RankType.STARS);
+
+    // Test suite: https://regexr.com/7e5gr
+    private static final Pattern GUILD_MESSAGE_PATTERN = Pattern.compile("§3\\[(?:§b)?★{0,5}(?:§3)?.{1,16}]§b");
+
+    // Test suite: https://regexr.com/7e66m
+    private static final Pattern RECRUIT_USERNAME_PATTERN = Pattern.compile("§3\\[(.{1,16})");
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void onChatMessageReceived(ChatMessageReceivedEvent e) {
+        if (rankType.get() == RankType.STARS) return; // default wynn behavior is stars
+
+        StyledText styledText = e.getStyledText();
+
+        Matcher m = styledText.getMatcher(GUILD_MESSAGE_PATTERN);
+        if (!m.find()) return;
+
+        MutableComponent modifiedComponent;
+        if (rankType.get() == RankType.NONE) {
+            modifiedComponent = getModifiedRemovedComponent(styledText);
+        } else {
+            modifiedComponent = getModifiedNamedComponent(styledText);
+        }
+
+        if (modifiedComponent.equals(styledText.getComponent())) return; // no changes
```

Comment:
The thing is, comparing components might not be stable enough. Please return the styled text and compare that (where we most likely end up with a correct equals check). Sorry for the miscommunication from my earlier comment.